src/equal.c

```diff
@@ -137,14 +137,49 @@ static int cpl_equal_scalar(const Rcomplex* x, const Rcomplex* y, bool na_equal)
     return real_equal && imag_equal;
   }
 }
+
+// The minimal number of checks required to determine if we need to translate.
+// - CHAR_ENC_TYPE - Don't translate these cases:
+//   (utf8 + utf8), (latin1 + latin1), (unknown + unknown), (bytes + bytes),
+//   (unknown + bytes)
+// - CHAR_IS_BYTES - Don't translate these cases:
+//   (bytes + utf8), (bytes + latin1)
+// - Translate these:
+//   (utf8 + latin1), (unknown + utf8), (unknown + latin1)
```

Comment:
Do we need to translate unknown + utf8 and unknown + latin1? Doesn't unknown imply ASCII?

Reply:
I think unknown+utf8 was the original error right? https://github.com/r-lib/vctrs/issues/553#issuecomment-526650247

Reply:
Ah, I think I always get confused about how this works — you can't change the encoding of an ASCII string (it's always unknown), but you can give a UTF-8 string unknown encoding.